handler.go

```diff
@@ -123,6 +127,15 @@ func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func withDefaultVersion(version string, path string) string {
+	if version == "" || versionRegexp.MatchString(path) {
+		return path
+	}
+	return version + "/" + path
+}
+
+var versionRegexp = regexp.MustCompile(`^v\d`)
```

Comment:
nit: might be clearer to move this above the function above

There's probably a corner-case for `v1`, which should not be matched

Reply:
Why shouldn't `v1` be matched? I think if  there is any `v\d` at the start of the path we don't want to prepend a different version.

Reply:
Train of through here was that Golang only uses `/vXX` for imports > v2, so v1 packages would be named without v1 in their name

Reply:
Right. So with this change it would be impossible to get redirected to the v1 pages on http://pkg.go.dev/. It assumes that anyone typing in the url wants the latest version if there is no version in the path. You can still go to the pre-v3 pages once you get there.

I don't see a way around that. I'd rather not add more logic for redirecting `v1`, since that is unlikely to be useful.